lib/Elastica/Reindex.php

```diff
@@ -32,84 +46,94 @@ class Reindex
      */
     protected $_options;
 
-    public function __construct(Index $oldIndex, Index $newIndex, array $options = [])
+    /**
+     * @var Response|null
+     */
+    protected $_lastResponse;
+
+    public function __construct(Index $oldIndex, Index $newIndex, array $params = [])
     {
         $this->_oldIndex = $oldIndex;
         $this->_newIndex = $newIndex;
-        $this->_options = $options;
+
+        $this->setParams($params);
     }
 
     public function run()
     {
-        $body = $this->_getBody($this->_oldIndex, $this->_newIndex, $this->_options);
+        $body = $this->_getBody($this->_oldIndex, $this->_newIndex, $this->getParams());
 
         $reindexEndpoint = new \Elasticsearch\Endpoints\Reindex();
+        $params = \array_intersect_key($this->getParams(), \array_fill_keys($reindexEndpoint->getParamWhitelist(), null));
+        $reindexEndpoint->setParams($params);
         $reindexEndpoint->setBody($body);
 
-        $this->_oldIndex->getClient()->requestEndpoint($reindexEndpoint);
-        $this->_newIndex->refresh();
+        $this->_lastResponse = $this->_oldIndex->getClient()->requestEndpoint($reindexEndpoint);
 
-        return $this->_newIndex;
+        return $this->_lastResponse;
```

Comment:
+1 on returning the response. The _newIndex will still be accessible through the object but the response would be lost. Should we even store it in the object?

Reply:
Storing the Response is only useful for `getTaskId()`, which is not that important to me as it's easy to get now (thanks to the response being returned). I will remove this method.

Reply:
I missed that you use it for getTaskId. I must confess there could be some convenience for users that know less about reindex to have the getTaskId method available.